caas/kubernetes/provider/application/application.go

```diff
@@ -313,6 +314,28 @@ func (a *app) Ensure(config caas.ApplicationConfig) (err error) {
 					MountPath: mountPath,
 				}, nil
 			},
+			func(pvc corev1.PersistentVolumeClaim, pvcName string, volumeName string) error {
+				pvc.ObjectMeta.Name = pvcName
+				pvc.ObjectMeta.Namespace = a.namespace
+				pvc.Spec.VolumeName = volumeName
+
+				expectedAnnotations := pvc.ObjectMeta.Annotations
+				expectedLabels := pvc.ObjectMeta.Labels
```

Comment:
I should have mentioned this earlier - we can't check for exact label and annotation match because the model could have been upgraded from an older one without the same data being written. Just check for (I think) `LabelKubernetesAppManaged` and whether the model labels match by calling `LabelsForModel()` to get the labels to check for

Reply:
I check the pvc created without `--attach-storage` now, the labels are:

```yaml
  labels:
    app.kubernetes.io/managed-by: juju
    app.kubernetes.io/name: psql
    storage.juju.is/name: pgdata
```

The `LabelsForModel` is not included in current pvc.

Reply:
`LabelsForModel()` is not a label name but a method used to get the required labels for a given model, taking into account the label versions for that model. We should check that the storage name matches. The app could have been renamed when deployed so we can't really check that. But the storage name is important to match.